### PR TITLE
Np 4033/add permission to all column

### PR DESCRIPTION
**Pull request by yij793**
address: https://nanopay.atlassian.net/browse/NP-3495
notes: the column list also need permission to display
Screen Shot 2021-05-12 at 1 02 13 PM

### CODE REVIEW

File: src/foam/u2/view/UnstyledTableView.js [hunk 1]
```diff
@@ -317,6 +307,15 @@ foam.CLASS({
     async function initE() {
       var view = this;
 
+      this.currentMemento_ = null;
+
```
Comment (olhabn) [inline src/foam/u2/view/UnstyledTableView.js:311]: please delete line 310 if you don't need it

File: src/foam/u2/view/UnstyledTableView.js [hunk 2]
```diff
@@ -724,6 +723,13 @@ foam.CLASS({
       function returnMementoColumnNameDisregardSorting(c) {
         return c && this.shouldColumnBeSorted(c) ? c.substr(0, c.length - 1) : c;
       },
+      async function filterUnpermited(arr) {
+        if ( this.auth ) {
+          const results = await Promise.all(arr.map( async p => p.hidden? false: ! p.columnPermissionRequired || await this.auth.check(null, `${this.of.name}.column.${p.name}`)));
```
Comment (olhabn) [inline src/foam/u2/view/UnstyledTableView.js:728]: can you please add white space after `p.hidden` and before `?`

**Comment by olhabn** [comment]
LGTM

**Comment by olhabn** [comment]
Other then tiny comments, LGTM

**Review by sarthak-marwaha — COMMENTED** [review]
LGTM other than @olhabn 's comments

File: src/foam/u2/view/UnstyledTableView.js [hunk 3]
```diff
@@ -724,6 +721,13 @@ foam.CLASS({
       function returnMementoColumnNameDisregardSorting(c) {
         return c && this.shouldColumnBeSorted(c) ? c.substr(0, c.length - 1) : c;
       },
+      async function filterUnpermited(arr) {
+        if ( this.auth ) {
+          const results = await Promise.all(arr.map( async p => p.hidden ? false : ! p.columnPermissionRequired || await this.auth.check(null, `${this.of.name}.column.${p.name}`)));
```
Comment (Nauna) [inline src/foam/u2/view/UnstyledTableView.js:726]: 80 char lines please

**Review by Nauna — APPROVED** [review]

**Comment by kgrgreer** [comment]
reviewing...

File: src/foam/u2/view/UnstyledTableView.js [hunk 4]
```diff
@@ -317,6 +307,13 @@ foam.CLASS({
     async function initE() {
       var view = this;
 
+      const asyncRes = await this.filterUnpermited(view.of.getAxiomsByClass(foam.core.Property));
```
Comment (kgrgreer) [inline src/foam/u2/view/UnstyledTableView.js:310]: Correct spelling is "Unpermitted"

File: src/foam/u2/view/UnstyledTableView.js [hunk 5]
```diff
@@ -724,6 +721,16 @@ foam.CLASS({
...
       },
+      async function filterUnpermited(arr) {
+        if ( this.auth ) {
+          const results = await Promise.all(arr.map( async p => 
+            p.hidden ? false : 
+            ! p.columnPermissionRequired || 
+            await this.auth.check(null, `${this.of.name.toLowerCase()}.column.${p.name}`)));
+          return arr.filter((_v, index) => results[index]);
+        }
+        return arr
```
Comment (kgrgreer) [inline src/foam/u2/view/UnstyledTableView.js:732]: Add ;

**Pull request merged**